db/python/tables/participant.py

```diff
@@ -586,7 +587,11 @@ async def get_external_ids_by_participant(
         GROUP BY participant_id
         """
         rows = await self.connection.fetch_all(_query, {'pids': participant_ids})
-        return {r['id']: from_db_json(r['external_ids']) for r in rows}
+        retval: dict[int, list[str]] = {}
+        for row in rows:
+            external_ids = parse_sql_dict(row['external_ids'])
```

Comment:
No, this SQL uses JSON_ARRAYAGG not JSON_OBJECTAGG. I guess a useful clarification would be to change the field name to `… AS external_ids_list` in the SQL to signpost this difference from other queries.